content/articles/leandrocfe-navigating-filament-pages-with-previous-and-next-buttons.md

```diff
@@ -0,0 +1,191 @@
+
+## Implementing New Actions
+
+To kick things off, we need to introduce two new actions: PreviousAction and NextAction. These actions will reside in the `app/Filament/Resources/Actions` folder.
+
+**PreviousAction.php**
+
+```php
+//app/Filament/Resources/Actions/PreviousAction.php
+namespace App\Filament\Resources\Actions;
+
+use Filament\Actions\Action;
+
+class PreviousAction extends Action
+{
+    public static function getDefaultName(): ?string
+    {
+        return 'previous';
+    }
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->hiddenLabel()
+            ->icon('heroicon-o-arrow-left')
+            ->outlined()
+            ->tooltip("Previous {$this->getRecordTitle()}");
+    }
+}
+```
+
+**NextAction.php**
+
+```php
+//app/Filament/Resources/Actions/NextAction.php
+namespace App\Filament\Resources\Actions;
+
+use Filament\Actions\Action;
+
+class NextAction extends Action
+{
+    public static function getDefaultName(): ?string
+    {
+        return 'next';
+    }
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->hiddenLabel()
+            ->icon('heroicon-o-arrow-right')
+            ->outlined()
+            ->tooltip("Next {$this->getRecordTitle()}");
+    }
+}
+```
+
+## Configuring Actions in Pages
+
+Next, we extend the capabilities of the ViewRecord and EditRecord classes by adding a `CanPaginateViewRecord` trait. This trait, residing in the `app/Filament/Resources/Pages/Concerns` folder, configures actions for pagination and provides methods to retrieve previous and next records based on the current record.
+
+```php
+namespace App\Filament\Resources\Pages\Concerns;
+
+use App\Filament\Resources\Actions\NextAction;
+use App\Filament\Resources\Actions\PreviousAction;
+use Filament\Actions\Action;
+use Filament\Actions\DeleteAction;
+use Filament\Actions\EditAction;
+use Filament\Actions\ForceDeleteAction;
+use Filament\Actions\ReplicateAction;
+use Filament\Actions\RestoreAction;
+use Illuminate\Database\Eloquent\Model;
+
+trait CanPaginateViewRecord
+{
+    protected function configureAction(Action $action): void
+    {
+        $this->configureActionRecord($action);
+
+        match (true) {
+            $action instanceof DeleteAction => $this->configureDeleteAction($action),
+            $action instanceof EditAction => $this->configureEditAction($action),
+            $action instanceof ForceDeleteAction => $this->configureForceDeleteAction($action),
+            $action instanceof ReplicateAction => $this->configureReplicateAction($action),
+            $action instanceof RestoreAction => $this->configureRestoreAction($action),
+
+            //new actions
+            $action instanceof PreviousAction => $this->configurePreviousAction($action),
+            $action instanceof NextAction => $this->configureNextAction($action),
+            
+            default => null,
+        };
```

Comment:
Since this `configureAction()` method is overriding the `configureAction()` method on the page that you're adding these pagination buttons to, could you leverage the existing parent method so that you don't have to write out all of the match statements for `DeleteAction`, `EditAction`, etc.?

Something like:

```php
match (true) {
    $action instanceof PreviousAction => $this->configurePreviousAction($action),
    $action instanceof NextAction => $this->configureNextAction($action),
    default => parent::configureAction($action),
};
```

Not sure if this would work, but it would clean up this code a bit and make it so that if any _other_ built-in actions (like `DeleteAction` and `EditAction`) were added, you wouldn't have to keep up with those changes and add them to this trait.